### ISSUE TITLE
Rust Analyzer fails to parse macro invocation correctly, leftover tokens

### ISSUE THREAD

**Issue by PhilipDaniels** [rust-lang/rust-analyzer#9956]
I have a crate https://crates.io/crates/logging_timer/1.0.0 which contains a macro which can be invoked like this:

```
let _tmr = stimer!(Level::Info; "Some string to log");
```

RA barfs on this, it says 'leftover tokens'. Seems to be getting confused by the semicolon after `Info`. This is valid Rust syntax.

(FWIW I can't quite remember why I did it that way, but it was probably because I was having trouble writing matching rules that correctly disambiguated all the ways this macro can be invoked. I do not know much about macros. In any case, I don't think it should generate an error).

**Comment by Veykril** [comment]
Seems to work now
